From 947a4e6098b4f81c3366ba20164a698e01a9e33d Mon Sep 17 00:00:00 2001
From: allen yan <yanwei@marvell.com>
Date: Fri, 1 Dec 2017 14:52:30 +0800
Subject: [PATCH 0711/1200] dts: a3700: Add vbus regulator supply to usb2

The USB device should linked to VBUS regulator through
"vbus-supply" DTS property.
This patch adds the "fixed-regulator" node for the USB2 VBUS
regulator.
This patch adds handling for "vbus-supply" property inside
the USB2 node in armada-3720-db.dts for turning on/off the
VBUS regulator.

Change-Id: Icc09c2f90530a3b473dbbc598be6e95a4d53c634
Signed-off-by: allen yan <yanwei@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/47121
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/55111
Reviewed-by: Hua Jing <jinghua@marvell.com>
Tested-by: Hua Jing <jinghua@marvell.com>
---
 arch/arm/dts/armada-3720-db.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/dts/armada-3720-db.dts b/arch/arm/dts/armada-3720-db.dts
index 814dc49dae..caf2a687b4 100644
--- a/arch/arm/dts/armada-3720-db.dts
+++ b/arch/arm/dts/armada-3720-db.dts
@@ -75,8 +75,19 @@
 		shutdown-delay-us = <1000000>;
 		gpio = <&expander0 1 GPIO_ACTIVE_HIGH>;
 	};
+
+	reg_usb2_vbus: usb2_vbus@0 {
+		compatible = "regulator-fixed";
+		regulator-name = "usb2-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		shutdown-delay-us = <1000000>;
+		gpio = <&expander0 0 GPIO_ACTIVE_HIGH>;
+	};
+
 };
 
+
 &comphy {
 	phy0 {
 		phy-type = <COMPHY_TYPE_PEX0>;
@@ -166,6 +177,7 @@
 
 /* CON29 */
 &usb2 {
+	vbus-supply = <&reg_usb2_vbus>;
 	status = "okay";
 };
 
-- 
2.22.0

